# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1338938477 25200
#      Tue Jun 05 16:21:17 2012 -0700
# Branch stable
# Node ID c50cc3c5089070cd382e190cd5f5ca8df6546b05
# Parent  b9bbd4c569baffdbfefe6ffe7ac44108dbd16f62
prctile.m: Fix error when called with just a single argument (bug #36597).

* prctile.m: Fix error when called with just a single argument (bug #36597).

diff --git a/scripts/statistics/base/prctile.m b/scripts/statistics/base/prctile.m
--- a/scripts/statistics/base/prctile.m
+++ b/scripts/statistics/base/prctile.m
@@ -54,17 +54,17 @@ function q = prctile (x, p = [], dim)
     p = [0, 25, 50, 75, 100];
   endif
 
   if (! (isnumeric (p) && isvector (p)))
     error ("prctile: P must be a numeric vector");
   endif
 
   nd = ndims (x);
-  if (nargin == 2)
+  if (nargin < 3)
     if (nd == 2)
       ## If a matrix or vector, always use 1st dimension.
       dim = 1;
     else
       ## If an N-d array, find the first non-singleton dimension.
       (dim = find (sz > 1, 1)) || (dim = 1);
     endif
   else
